bundles/core/src/main/java/com/adobe/cq/commerce/core/components/services/UrlProvider.java

```diff
@@ -100,30 +107,59 @@
     public String toProductUrl(SlingHttpServletRequest request, @Nullable Page page, Map<String, String> params);
 
     /**
-     * Returns the category page URL.
+     * Returns the product page URL. Only the product identifier must be provided, the implementation will query the needed URL
+     * attributes to generate a complete URL based on the configuration.
+     * 
+     * This method should be used if the component only can provide the product identifier.
+     * 
+     * @param request The current Sling HTTP request.
+     * @param page The target page, if any. This parameter can be null if the URL template does not use the <code>${page}</code> parameter.
+     * @param productIdentifier The product identifier.
+     * @return The product URL.
+     */
+    public String toProductUrl(SlingHttpServletRequest request, @Nullable Page page, String productIdentifier);
```

Comment:
While we are on it: would it make sense to introduce short-hand api for those as well (omitting the nullable arguments)?

`toProductUrl(SlingHttpServletReques, String productIdentifier)`